apps/avifenc.c

```diff
@@ -739,6 +751,16 @@ int main(int argc, char * argv[])
     }
     avifBool sourceWasRGB = (inputFormat != AVIF_APP_FILE_FORMAT_Y4M);
 
+    // check again for y4m input (y4m input decide yuv format by file but not command arguments)
```

Comment:
I don't understand this change. By the time we get here, the avifInputReadImage() call at line 746 has already called avifImageRGBToYUV() using image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_IDENTITY, so the damage is already done. Isn't it too late to change image->matrixCoefficients to AVIF_MATRIX_COEFFICIENTS_BT601?

Reply:
I see you already described this change:

`I changed to check input.requestedFormat, which holds the value specified by user. However y4m input doesn't use this setting so we have to check again. There's still the case that a yuv444 y4m input with --cicp 1/13/0 -y 420 will make avifenc wrongly change the MC, but as the user is doing wrong things I think it's OK.`

I think the right thing to do in this case is to exit the program with an error message, rather than encoding the image with an incorrect MC.

Also, this seems like a pre-existing problem, not introduced by your PR. You merely moved this block of code forward a few lines.

Reply:
Done